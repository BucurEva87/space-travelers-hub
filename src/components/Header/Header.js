import React from 'react';
import Navigation from '../Navigation/Navigation';
import style from './Header.module.scss';

const Header = () => (
  <header>
    <a href="/" className={style.logo}>
      <img
        src={`${process.env.PUBLIC_URL}images/logo.png`}
        alt="Space travelers hub logo"
      />
      <h1>Space Travelers Hub</h1>
    </a>
    <Navigation />
  </header>
);

export default Header;
