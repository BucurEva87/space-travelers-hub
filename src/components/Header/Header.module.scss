@import '../../styles/variables';

header {
  width: 100%;
  display: flex;
  align-items: center;
  border-bottom: 0.1rem solid $gray;
  margin-bottom: 1.2rem;
}

.logo {
  display: flex;
  align-items: center;
  gap: 1rem;
  padding-top: 0.8rem;
  padding-bottom: 1.6rem;
}

img {
  width: 2rem;
}
