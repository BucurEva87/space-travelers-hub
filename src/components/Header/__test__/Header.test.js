import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../Header';

describe('Header component', () => {
  test('renders on the screen', () => {
    expect(render(
      <Router>
        <Header />
      </Router>,
    ).getByTestId('header')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Router>
        <Header />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
