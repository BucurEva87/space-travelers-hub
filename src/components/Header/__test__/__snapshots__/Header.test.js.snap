// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<header
  data-testid="header"
>
  <a
    className="logo"
    href="/"
  >
    <img
      alt="Space travelers hub logo"
      src="images/logo.png"
    />
    <h1>
      Space Travelers Hub
    </h1>
  </a>
  <nav
    className="nav"
    data-testid="navbar"
  >
    <ul>
      <li>
        <a
          className=""
          href="/rockets"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className=""
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className=""
          href="/dragons"
          onClick={[Function]}
        >
          Dragons
        </a>
      </li>
      <li>
        <a
          className=""
          href="/myprofile"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
